app/src/main/java/com/teamagam/gimelgimel/app/model/entities/messages/InMemory/InMemoryMessagesModel.java

```diff
@@ -0,0 +1,43 @@
+package com.teamagam.gimelgimel.app.model.entities.messages.InMemory;
+
+import com.teamagam.gimelgimel.app.common.NotifyingDataChangedObservable;
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+import com.teamagam.gimelgimel.app.model.entities.messages.MessagesModel;
+
+import java.util.ArrayList;
+
+/**
+ * {@link MessagesModel} class that holds its messages in-memory
+ */
+public class InMemoryMessagesModel extends NotifyingDataChangedObservable implements MessagesModel {
+
+    private ArrayList<Message> mMessages = new ArrayList<>();
```

Comment:
They dont need to be singletons. We'll only create one instance of them. !resolved